base/src/main/java/io/spine/reflect/Invokables.java

```diff
@@ -0,0 +1,211 @@
+                        method.getName(), target,
+                        target.getClass()
+                              .getCanonicalName()));
+        return result;
+    }
+
+    /**
+     * Tries to find a constructor with no arguments in the specified class or its parents.
+     *
+     * <p>If no such constructor has been found, throws an {@code IllegalArgumentException}.
+     *
+     * @param type
+     *         class to look for constructors in
+     * @return a constructor with no parameters, if it exists
+     * @throws IllegalArgumentException
+     *         if the specified class does not declare a parameterless constructor
+     */
+    @CanIgnoreReturnValue
+    private static <C> Constructor<C> ensureParameterlessCtor(Class<C> type) {
+        checkNotNull(type);
+        @SuppressWarnings("unchecked" /* safe, as `Class<C>` only declares `Constructor<C>`. */)
+        Constructor<C>[] ctors = (Constructor<C>[]) type.getDeclaredConstructors();
+        for (Constructor<C> ctor : ctors) {
+            if (ctor.getParameterCount() == 0) {
+                return ctor;
+            }
+        }
+
+        throw newIllegalArgumentException("No parameterless ctor found in class `%s`.",
+                                          type.getSimpleName());
+    }
+
+    /**
+     * Performs a reflective operation regardless of its accessibility, returns its result.
+     *
+     * <p>Upon completion, whether successful or erroneous, returns the accessibility to its
+     * initial state.
+     *
+     * @param reflectiveObject
+     *         an object that can be used to make an {@code Invokable}
+     * @param makeInvokable
+     *         a function of {@code P} -> {@code Invokable}. {@code Invokable} is
+     *         needed to manipulate the accessibility in a generic manner
+     * @param fn
+     *         a reflective function to perform
+     * @param onError
+     *         a supplier of the error message to include into the {@code
+     *         IllegalStateException} should an error be thrown
+     * @param <T>
+     *         a type of reflection-related object to perform a function on
+     * @param <R>
+     *         a type of result of the reflective function
+     * @return a result of the reflective function
+     * @throws IllegalStateException
+     *         if a {@code ReflectiveOperationException} is thrown
+     *         by {@code fn}, or other error occurs during the reflective operation execution
+     */
+    @SuppressWarnings("OverlyBroadCatchBlock" /* catching any runtimes does not hurt here. */)
+    private static <T, R> R invokePreservingAccessibility(T reflectiveObject,
+                                                          Function<T, Invokable<?, ?>> makeInvokable,
+                                                          ReflectiveFunction<T, R> fn,
+                                                          Supplier<String> onError) {
+        checkNotNull(reflectiveObject);
```

Comment:
We can probably save on null checks, assuming this is a private method. By the type we get here outer API should have checked all the arguments.